storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfigTest.java

```diff
@@ -105,4 +116,79 @@ private Map<String, Object> extractProps(RemoteLogManagerConfig remoteLogManager
                   remoteLogManagerConfig.remoteLogMetadataManagerPrefix());
         return props;
     }
+
+    @Test
+    public void testHashCode_ForAllAndTwoFields() {
+        String rsmPrefix = "__custom.rsm.";
+        String rlmmPrefix = "__custom.rlmm.";
+        Map<String, Object> rsmProps = Collections.singletonMap("rsm.prop", "val");
+        Map<String, Object> rlmmProps = Collections.singletonMap("rlmm.prop", "val");
+        RemoteLogManagerConfig config1 = getRemoteLogManagerConfig(false, rsmPrefix, rlmmPrefix, rsmProps, rlmmProps);
+        RemoteLogManagerConfig config2 = getRemoteLogManagerConfig(false, rsmPrefix, rlmmPrefix, rsmProps, rlmmProps);
+
+        // Initially, hash codes should be equal for default objects
+        assertEquals(config1.hashCode(), config2.hashCode());
+
+        // Test for specific field remoteLogManagerCopierThreadPoolSize
+        RemoteLogManagerConfig config3 = new RemoteLogManagerConfig(true, "dummy.remote.storage.class",
```

Comment:
Could you also add a builder to the RemoteLogManagerConfig class? It's difficult to read which value is for which parameter.